### DISCUSSION TITLE
Mediation pathways and sample weights

### DISCUSSION THREAD

**Discussion by sobhneek** [raphaelvallat/pingouin#350] [Q&A] [answered]
Hi!

**Mediation Pathways:**
- So I was a little unsure on how some of the pathways interact with each other in the mediation function, particularly when covariates and a second mediator are thrown into the mix. I attached a picture below of a path diagram and I was wondering if the pathways that I made are representative of the code below. 

**Sample Weights:**
- Since there isn't a built in parameter for sample weights, would passing something through `logreg_kwargs` (see code below) work?  

Thank you.

```Python
pg.mediation_analysis(data= df,
                      x='X’,
                      m=[‘M1’,’M2’],
                      y=‘Y’,
                      covar=[‘C1’,’C2’],
                      logreg_kwargs={'sample_weights':'Weights'})
```

<img width="900" alt="Screenshot 2023-03-08 at 1 13 34 PM" src="https://user-images.githubusercontent.com/98254394/223795857-93f775be-668c-4e2f-8d21-1ee0ed23257c.png">

**Comment by raphaelvallat** [comment, marked as answer]
Hi @sobhneek,

One of the best reference I know on the topic of mediation is this paper: https://www.sciencedirect.com/science/article/abs/pii/S0005796716301887?via%3Dihub

They recommend adding covariates to all the regressions in the mediation pathway, which is what Pingouin does. They also report a path diagram with covariates included for a single-mediator model:

<img width="441" alt="image" src="https://user-images.githubusercontent.com/25703792/223967557-5c44ed40-b59e-43af-85d0-7a40fc2dc0a6.png">


Regarding the `sample_weights`, this will not work because the `logreg_kwargs` are passed to the LogisticRegression estimator but not to the `.fit` [method](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression.fit):

https://github.com/raphaelvallat/pingouin/blob/c22843f6f8f243d13e4bc82a0644763aedab02bf/pingouin/regression.py#L890-L898

**Reply by sobhneek** [reply]
Hi @raphaelvallat 

Thank you for the insight. I guess my only question now is if you wanted to include sample weights, how would you go about doing so?

**Reply by raphaelvallat** [reply]
To be honest, I don't know if the mediation results would be valid with sample weights. But, the easiest way to do this is to clone the Pingouin repository, and then edit the code to add an optional `kwargs_fit` parameter that will be passed to LogisticRegression.fit(**)
